storage/blob.go

```diff
@@ -513,6 +513,57 @@ func (b BlobStorageClient) GetBlob(container, name string) (io.ReadCloser, error
 	return resp.body, nil
 }
 
+// GetBlob returns a stream to read the blob. Caller must call Close() the
+// reader to close on the underlying connection.
+//
+// See https://msdn.microsoft.com/en-us/library/azure/dd179440.aspx
+func (b BlobStorageClient) GetBlobAndProperties(container, name string) (io.ReadCloser, *BlobProperties, error) {
```

Comment:
you should probably write an unexported helper method that does this, and use that in GetBlobAndProperties (directly) as well as in GetBlob (by discarding BlobProperties). Otherwise it's duplicating code for no good reason.